docs/settings/reporting-settings.asciidoc

```diff
@@ -27,8 +27,21 @@ setting to preserve the same key across multiple restarts and multiple instances
 ==== Kibana server settings
 
 Reporting uses the Kibana interface to generate reports. In most cases, you don't need
-to configure Reporting to communicate with Kibana. However, if you use a reverse-proxy
-to access Kibana, you must set the proxy port, protocol, and hostname.
+to configure Reporting to communicate with Kibana. If you use a reverse-proxy to access Kibana, you can set the proxy port, protocol, 
+and hostname to have Reporting use the proxy to load Kibana URLs. In most cases, 
+this is only required if the proxy is responsible for modifying requests to 
+make them work with Kibana. 
+
+For example, a reverse proxy can be responsible for translating the public hostname 
+for the Kibana server to a different name in a private network. 
+It can also be responsible for maintaining encrypted communications between 
+the outside world and hosts in the internal network. 
```

Comment:
In these changes, I think everything up to this part is true, and helpful:

> For example, a reverse proxy can be responsible for translating the public hostname for the Kibana server to a different name in a private network.

But where it goes from there is more of technical considerations to make AFTER making the change to set a `xpack.reporting.kibanaServer.*` to get connections working end-to-end.

@elastic/kibana-operations would you be able to help out here? It's hard to articulate the problem these reporting settings are solving.

1. Problem: Imagine a case where Kibana server listens on a particular port that is only exposed to a proxy. The Reporting headless browser starts and tries to connect to that server, but the connection can not be made because just using say, `localhost:5601` wouldn't work.

2. Solution: Set up the `xpack.reporting.kibanaServer.*` settings to point to the proxy that Kibana talks to. The connection would succeed if the headless browser connects to the proxy that reverses the connections back to the actual Kibana server.

3. Potential problem with the solution: We set `ignore-ssl-errors` to true when launching Chromium, but when the admin changes Reporting to use the proxy, our attempt to ignore SSL errors stops working, because connections with SSL errors will be terminated by the proxy.

4. Advice for the potential problem: Therefore when using `xpack.reporting.kibanaServer.*` settings, they must be set to fully qualified values where encrypted communication will work end-to-end between the headless browser to the Kibana server, wherever it is on the network.

Reply:
I think I've been overthinking this, because this section does not need to be very long. How about:

> Reporting opens the Kibana web interface in a server process to generate screenshots of Kibana visualizations. In most cases, the default settings will work and you don't need to configure Reporting to communicate with Kibana. However, if your client connections must go through a reverse-proxy
to access Kibana, Reporting configuration must have the proxy port, protocol, and hostname set in the `xpack.reporting.kibanaServer.*` settings.


> **NOTE**: If a reverse-proxy carries encrypted traffic from end-user clients back to a Kibana server, the proxy port, protocol, and hostname in Reporting settings must be valid for the encryption that the Reporting browser will receive. Encrypted communications will fail if there are mismatches in the host information between the request and the certificate on the server.

Reply:
Hopefully the NOTE would scare people away from using these settings unless it is necessary, which is something this PR is trying to improve.